src/intervals/functions.jl

```diff
@@ -373,3 +373,13 @@ function nthroot(a::Interval{T}, n::Integer) where T
     b = nthroot(bigequiv(a), n)
     return convert(Interval{T}, b)
 end
+
+"""
+Calculate `x mod y` where `x` is an interval and `y` is a positive divisor.
+"""
+function mod(x::Interval, y::Real)
+    @assert y > zero(y) "modulo is currently implemented only for a positive divisor."
```

Comment:
Is it possible to extend this to have strictly negative `y`? I understand that having `0 ∈ y` complicates things....

Reply:
Updated for strictly negative divisor. Hope its correct.

Reply:
The implementation seems to me correct if we have `y::AbstractFloat` (perhaps also including Rationals, but let me forget about them now), which corresponds to the tests.

However, if `y::Interval` (and we have `Interval <: Real`) then there this function throws errors: e.g., try `mod(1..2, 1..2)`, which I think should return `Interval(0.0, 2.0)`. In this case things are subtle, because `y != zero(y)` is true for `[-1, 1]`, but that interval is not strictly positive or negative. Also, `Interval(zero(y), y)` causes the error mentioned above. I guess this is the reason that `mod` has two methods in #178.

My suggestion is either restrict `y::AbstractFloat`, or include a new method where `y::Interval`.

Reply:
This is my mistake. My intention was to constrain `y` NOT to be `Interval`. I've not realized `1..3 isa Real -> true`. However, If I constraint it to `y::AbstractFloat` it does not accept integers for example because `1 isa AbstractFloat -> false`, right?

Reply:
`mod` can indeed be generalized for `y::Interval`. It's tricky with respect of having zero within the interval, which is part of the reason I was suggesting to have either a strictly positive or negative `y`. Actually, a motivating example would be to have `y::Irrational`, or actually any (mathematical) real number which is not exactly representable as a `Float64`. Note that #178 includes such an implementation, though it does not include the restriction of strictly positive/negative intervals. The subtleties related to zero at the end are related to the division: `y` appears in the denominator.

Reply:
I unresolved this conversation, so it is easy to track the discussion...

Reply:
fwiw in octave

```
>> mod(infsup(-3, 2), infsup(-1, 0.5))
ans = [-1, +0.5]

>> mod(infsup(-3, 2), infsup(0, 0))
ans = [Empty]

>> mod(infsup(-3, 2), infsup(0, 3))
ans = [0, 3]

>> mod(infsup(-3, 2), infsup(1, 3))
ans = [0, 3]

>> mod(infsup(-3, 2), infsup(-3, 2))
ans = [-3, +2]
```